jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -1,658 +1,169 @@
-        })
-    outs = read_var_int()
-    for i in range(outs):
-        obj["outs"].append({
-            "value": read_as_int(8),
-            "script": hex_string(read_var_string(), hexout)
-        })
-    #segwit flag is only set if at least one txinwitness exists,
-    #in other words it would have to be at least partially signed;
-    #and, if it is, the witness section must be properly created
-    #including "00" for any input that either does not YET or will not
-    #have a witness attached.
-    if segwit:
-        #read witness data
-        #there must be one witness object for each txin
-        #technically, we could parse the contents of the witness
-        #into objects, but we'll just replicate the behaviour of the
-        #rpc decoderawtx, and attach a "txinwitness" for each in, with
-        #the items in the witness space separated
-        for i in range(ins):
-            num_items = read_var_int()
-            items = []
-            for ni in range(num_items):
-                items.append(hex_string(read_var_string(), hexout))
-            obj["ins"][i]["txinwitness"] = items
+import json
+# needed for single sha256 evaluation, which is used
+# in bitcoin (p2wsh) but not exposed in python-bitcointx:
+import hashlib
 
-    obj["locktime"] = read_as_int(4)
-    return obj
-
-def serialize(tx):
-    """ Accepts a dict in which the "outpoint","hash"
-    "script" and "txinwitness" entries can be in binary
-    or hex. If any "hash" field is in hex, the serialized
-    output is in hex, otherwise the serialization output
-    is in binary. Below table assumes all hex.
-    Table of dictionary keys and type for the value:
-    ================================================
-    version: int
-    ins: list
-    ins[0]["outpoint"]["hash"]: hex encoded string
-    ins[0]["outpoint"]["index"]: int
-    ins[0]["script"]: hex encoded string
-    ins[0]["sequence"]: int
-    ins[0]["txinwitness"]: list (optional, may not exist)
-    ins[0]["txinwitness"][0]: hex encoded string
-    outs: list
-    outs[0]["script"]: hex encoded string
-    outs[0]["value"]: int
-    locktime: int
-    =================================================
-    Returned serialized transaction is a byte string,
-    or a hex encoded string, according to the above
-    hash check.
+from jmbitcoin.secp256k1_main import *
+from jmbase import bintohex, utxo_to_utxostr
+from bitcointx.core import (CMutableTransaction, Hash160, CTxInWitness,
+                            CMutableOutPoint, CMutableTxIn, CTransaction,
+                            CMutableTxOut, CTxIn, CTxOut, ValidationError)
+from bitcointx.core.script import *
+from bitcointx.wallet import (P2WPKHCoinAddress, CCoinAddress, P2PKHCoinAddress,
+                              CCoinAddressError)
+from bitcointx.core.scripteval import (VerifyScript, SCRIPT_VERIFY_WITNESS,
+                                       SCRIPT_VERIFY_P2SH, SIGVERSION_WITNESS_V0)
+
+def hrt(tx, jsonified=True):
```

Comment:
Please rename `hrt()`, `hrinp()` and `hrout()` to something more descriptive. How about `human_readable_tx()` `human_readable_input()` `human_readable_output()`

Reply:
I agree with @chris-belcher here.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/536/commits/19249f7324910c20e9c76ddb3172500520d6d78b